packages/static/kairos-overlay-files/files/system/oem/52_installer.yaml

```diff
@@ -1,43 +1,31 @@
 name: "Start installer on tty1"
 stages:
   initramfs:
-    - if: grep -q "nodepair.enable" /proc/cmdline && [ "$(kairos-agent state get kairos.init)" == "systemd" ]
-      commands:
-        - systemctl disable getty@tty1
-        - systemctl stop getty@tty1
-        - systemctl mask getty@tty1
-        - systemctl enable kairos
-        - systemctl enable kairos-webui
-    - if: |
-        grep -q "nodepair.enable" /proc/cmdline && [ -f /run/cos/uki_install_mode ] && [ "$(kairos-agent state get kairos.init)" == "systemd" ]
+    # Start installer if we get the install-mode cmdline and NOT the install-mode-interactive cmdline as we can have both
+    # and we dont want to trigger both
+    # Works in livecd or uki install mode
+    - if: grep -q "install-mode" /proc/cmdline && grep -vq "install-mode-interactive" && ([ "$(kairos-agent state get boot)" == "livecd_boot" ] || [ -f /run/cos/uki_install_mode ]) && [ "$(kairos-agent state get kairos.init)" == "systemd" ]
       commands:
         - systemctl disable getty@tty1
         - systemctl stop getty@tty1
         - systemctl mask getty@tty1
         - systemctl enable kairos
         - systemctl enable kairos-webui
     # Starts installer on boot for openRC based systems
-    - if: grep -q "nodepair.enable" /proc/cmdline && [ "$(kairos-agent state get kairos.init)" == "openrc" ]
+    - if: grep -q "install-mode" /proc/cmdline && grep -vq "install-mode-interactive" && [ "$(kairos-agent state get kairos.init)" == "openrc" ]
```

Comment:
I think this produces a false positive `-v` will be successful whether "install-mode-interactive" is present or not

Reply:
why? Seems to work here:

```
# itxaka @ zeus in ~/projects/packages on git:nodepair_enable x [15:42:41] C:1
$ cat install.txt  
install-mode

# itxaka @ zeus in ~/projects/packages on git:nodepair_enable x [15:42:44] 
$ cat install-interactive.txt 
install-mode install-mode-interactive

# itxaka @ zeus in ~/projects/packages on git:nodepair_enable x [15:42:47] 
$ (grep -q install-mode install.txt && grep -vq install-mode-interactive install.txt)           

# itxaka @ zeus in ~/projects/packages on git:nodepair_enable x [15:42:57] 
$ echo $?
0

# itxaka @ zeus in ~/projects/packages on git:nodepair_enable x [15:42:58] 
$ (grep -q install-mode install-interactive.txt && grep -vq install-mode-interactive install-interactive.txt)

# itxaka @ zeus in ~/projects/packages on git:nodepair_enable x [15:43:06] C:1
$ echo $?
1
```

install.txt represents the cmdline of the defaul;t install mode, the cmdline that is on kairos by defaultt
install.txt represents the cmdline of uki with interactive install, which has both install-mode and install-mode-interactive

We want to only trigger the kairos install service if install-mode is in the cmdline but NOT install-mode-interactive, as for that the interactive installer service should trigger instead.


Where are you seeing the issue @mauromorales ? I cant find it and my local tests seems to confirm this (hint: is how I got this line, by testing it locally to make sure it worked lmao)

Reply:
sorry, my bad, tested again and working as expected. I got confused